src/charwise.rs

```diff
@@ -972,8 +970,8 @@ impl State {
 
     #[inline(always)]
     #[allow(dead_code)]
-    pub fn set_base(&mut self, x: u32) {
-        self.base = x;
+    pub fn set_base(&mut self, x: NonZeroU32) {
+        self.base = Some(x);
```

Comment:
```suggestion
        debug_assert_ne!(x.get(), 0);
        self.base = Some(x);
```
`NonZeroU32` will never have a value of zero unless we employ unsafe `new_unchecked`.
However,  `set_base` cannot ensure this, so a debug assertion would be beneficial.

Reply:
> However, set_base cannot ensure this
That is not the role of this function, and assigning 0 to NonZeroU32 is UB anyway.

Reply:
I see. Thanks for your explanation.